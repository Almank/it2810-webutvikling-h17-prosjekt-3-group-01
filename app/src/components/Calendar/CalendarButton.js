import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

export class CalendarButton extends React.Component {

    backgroundC(){
        //If no color is sent in props.
        if (this.props.backgroundC === undefined)
            return {
            backgroundColor: '#FF9505',
        };
        return {
            backgroundColor: this.props.backgroundC,
        };
    }

    render(){
        return (
            <TouchableOpacity
                style={[styles.addAppButton, this.backgroundC(), styles.shadow]}
                onPress={this.props.onpress}>
                <Text style={styles.textButton}>{this.props.text}</Text>
            </TouchableOpacity>
        )
    }
}

const styles = StyleSheet.create({
    addAppButton: {
        marginTop: 20,
        marginLeft:'15%',
        marginRight:'15%',
        flexBasis: 40,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#FF9505',
        borderWidth: 1,
        borderColor: 'rgba(155,155,155,0.5)',
    },
    backgroundColorRed: {
        backgroundColor: 'red',
    },
    textButton: {
        fontSize:16,
        color: 'white',
    },
    shadow: {
        shadowColor: 'black',
        shadowOffset: {width:0, height: 2},
        shadowOpacity: 0.3,
        shadowRadius: 5,
    },
});
