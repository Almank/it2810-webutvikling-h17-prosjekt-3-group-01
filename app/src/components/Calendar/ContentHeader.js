/**
 * Created by almank on 14.10.2017.
 */
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export class ContentHeader extends React.Component {

    render(){
        return (
            <View style={styles.container}>
                <Text style={styles.item}>When</Text>
                <Text style={styles.item}>Title</Text>
                <Text style={styles.item}>About</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        marginTop: 20,
        flexDirection: 'row',
        justifyContent: 'center',
        flexWrap: 'wrap',
        borderBottomWidth:1.5,
        borderBottomColor: 'black',
    },
    item: {
        flexBasis:'33%',
        textAlign:'center',
        fontSize:18,
        marginBottom: 20,
    }
});
