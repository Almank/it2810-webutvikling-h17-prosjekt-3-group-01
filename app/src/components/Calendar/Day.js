import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

export class Day extends React.Component {
    render(){
        return(
            <View style={styles.container}>
                <Text>{this.props.dayName}</Text>
                <Text>{this.props.dayDate}</Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        flexBasis: '25%',
        backgroundColor: 'yellow',
        height:100,
    }
});
