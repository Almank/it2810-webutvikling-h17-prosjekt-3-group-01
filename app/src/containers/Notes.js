import React from 'react';
import { View, Text } from 'react-native';

export class Notes extends React.Component {
    render(){
        return(
            <Text>Notes</Text>
        )
    }
}