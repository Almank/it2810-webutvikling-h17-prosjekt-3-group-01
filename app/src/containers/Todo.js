import React from 'react';
import {TodoList} from "../components/TodoList/TodoList";

export class Todo extends React.Component {
    render(){
        return(
            <TodoList />
        )
    }
}