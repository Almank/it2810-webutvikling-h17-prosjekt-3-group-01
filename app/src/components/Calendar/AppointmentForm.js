/**
 * Created by almank on 14.10.2017.
 */
import React from 'react';
import {ContentHeader} from "./ContentHeader";
import { View, Text, TextInput, TouchableOpacity, Modal, StyleSheet } from 'react-native';

export class AppointmentForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            modalVisible: false,
        };
        this.submitInput = this.submitInput.bind(this);
    }

    //Send data to calendar on submit
    submitInput() {
        if (this.state.dateValue !== null && this.state.timeValue !== null && this.state.titleValue !== null){
            this.props.getValues([this.state.dateValue, this.state.timeValue, this.state.titleValue, this.state.textValue]);
        } else {
            this.setState({
                dateValue: new Date().toISOString().slice(0, 10),
                timeValue: null,
                titleValue: null,
                textValue: null,
            })
        }
    }

    //Hide/show form.
    setModalVisible(visible) {
        this.setState({modalVisible: visible});
    }

    render(){
        return (
            <View>
                <TouchableOpacity
                    onPress={() => {
                        this.setModalVisible(true)
                        }}
                        title='Press me'
                        style={[styles.addAppButton, styles.shadow]}>
                    <Text style={styles.textButton}>ADD APPOINTMENT</Text>
                </TouchableOpacity>
                <ContentHeader />
                <Modal
                    animationType="slide"
                    transparent={false}
                    visible={this.state.modalVisible}
                    onRequestClose={() => {alert("Form has been closed.")}}>
                    <View style={styles.formContainer}>
                        <View>
                            <Text style={styles.title}>Create new appointment</Text>
                        </View>
                        <Text htmlFor="date" style={styles.text}>Date - YYYY-MM-DD</Text>
                        <TextInput onChangeText={(dateValue) => this.setState({dateValue})}
                                   value={this.state.dateValue}
                                   placeholder={'YYYY-MM-DD'}
                                   style={[styles.inputField, styles.shadow]}/>
                        <Text htmlFor="time" style={styles.text}>Time - HH:MM</Text>
                        <TextInput onChangeText={(timeValue) => this.setState({timeValue})}
                                   value={this.state.timeValue}
                                   placeholder={'HH:MM'}
                                   style={[styles.inputField, styles.shadow]}/>
                        <Text htmlFor="title" style={styles.text}>Title</Text>
                        <TextInput onChangeText={(titleValue) => this.setState({titleValue})}
                                   value={this.state.titleValue}
                                   style={[styles.inputField, styles.shadow]}/>
                        <Text htmlFor="what" style={styles.text}>What</Text>
                        <TextInput onChangeText={(textValue) => this.setState({textValue})}
                                   value={this.state.textValue}
                                   style={[styles.inputField, styles.shadow]}/>
                        <TouchableOpacity onPress={() => {
                            this.setModalVisible(!this.state.modalVisible);
                            //To avoid getting stuck by having alert behind modal.
                            setTimeout(() => {
                                this.submitInput();
                            }, 100);

                            }}
                            title='Submit'
                            style={[styles.addAppButton, styles.shadow]}>
                            <Text style={styles.textButton}>Submit</Text>
                        </TouchableOpacity>
                        <TouchableOpacity onPress={() => {
                            this.setModalVisible(!this.state.modalVisible)
                        }}
                                          style={[styles.addAppButton, styles.shadow]}>
                            <Text style={styles.textButton}>Cancel</Text>
                        </TouchableOpacity>
                    </View>
                </Modal>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    addAppButton: {
        marginTop: 20,
        marginLeft:'15%',
        marginRight:'15%',
        flexBasis: 40,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#FF9505',
        borderWidth: 1,
        borderColor: 'rgba(155,155,155,0.5)',
    },
    textButton: {
        fontSize:16,
        color:'white',
    },
    formContainer: {
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'column',
        height:'100%',
    },
    inputField: {
        padding: 5,
        marginLeft: '15%',
        marginRight: '15%',
        flexBasis: 40,
        backgroundColor: 'white',
        borderWidth: 1,
        borderColor: 'rgba(155,155,155,0.5)',
    },
    shadow: {
        shadowColor: 'black',
        shadowOffset: {width:0, height: 2},
        shadowOpacity: 0.2,
        shadowRadius: 5,
    },
    text: {
        marginLeft: '15%',
        marginRight: '15%',
        marginBottom: 3,
        marginTop: 10,
    },
    title: {
        fontSize: 25,
        textAlign: 'center',
    }

});