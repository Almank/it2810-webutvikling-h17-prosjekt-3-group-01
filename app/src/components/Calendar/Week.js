import React from 'react';
import { View, StyleSheet } from 'react-native';
import {Day} from './Day';

export class Week extends React.Component {

    getDayName(i) {
        const dayNames = ['MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY', 'SUNDAY'];
        const firstTwoDays = ['TODAY', 'TOMORROW'];
        let weekDay = new Date().getDay();
        if (i < 2)
            return firstTwoDays[i];
        if (i+weekDay > 7)
            return dayNames[i+weekDay-7-1]; //Minus 7 days, and minus 1 since it starts at 0.
        return dayNames[i+weekDay-1];
    }

    getDate(i){
        return new Date().getDate() + i;
    }

    getFullDate(i){
        let date = new Date();
        date.setDate(date.getDate() + i)
        return date.toISOString().slice(0,10);
    }

    render(){

        let weekdays = [];
        for (let i=0; i<7; i++){
            weekdays.push(<Day  key={this.getFullDate(i)}
                                dateFull={this.getFullDate(i)}
                                dayName={this.getDayName(i)}
                                dayDate={this.getDate(i)} />);
        }
        return(
            <View style={styles.container}>
                {weekdays}
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
        flexWrap: 'wrap',
        backgroundColor: 'red',
        flex: 1,
    }
});
