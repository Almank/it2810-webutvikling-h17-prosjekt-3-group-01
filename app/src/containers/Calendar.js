import React from 'react';
import { View, StyleSheet, AsyncStorage, Text, ScrollView } from 'react-native';
import {Week} from '../components/Calendar/Week';
import {AppointmentForm} from "../components/Calendar/AppointmentForm";

export class Calendar extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            dateToday: new Date().toISOString().slice(0, 10),
            dateVal: '',
            timeVal: '',
            titleVal: '',
            textVal: '',
            children: [
                {date: '2017-10-15', time: '14:15', title: 'Hello World', text: 'This is text'},
            ],
        };

        //Binding functions
        this.createAppointment = this.createAppointment.bind(this);
        this.changeContent = this.changeContent.bind(this);

        this.loadData();
    }

    emptyScheduleCheck(){

        //Copy of the state children array.
        let myData =[].concat(this.state.children)

        //Sorting elements based on time, earliest first.
            .sort((a, b) => a.time > b.time)

            //Filtering out appointments on current day.
            .filter(child => child.date === this.state.dateToday)

            //Mapping items from array giving the html the correct values.
            .map((item,i) =>
                <View style={styles.showAppointments} key={i}>
                    <Text style={styles.appointmentItem}>{item.time}</Text>
                    <Text style={styles.appointmentItem}>{item.title}</Text>
                    <Text style={styles.appointmentItem}>{item.text}</Text>
                </View>
            );

        //If there are no plans that day.
        if (myData.length === 0){
            myData = [  <View style={styles.showAppointments} key={0}>
                <Text style={styles.appointmentNoItems}>You don't have any plans!</Text>
            </View>];
        }

        return (
            <ScrollView style={styles.scrollView}>
                {myData}
            </ScrollView>
        );
    }

    changeContent(e){
        let data = this.state;
        data.dateToday = e.props.dateFull;
        this.setState({ dateToday: e.props.dateFull }, () => {
            this.forceUpdate();
        });
        try{
            AsyncStorage.setItem("calendar", JSON.stringify(data));
        } catch (error){
            console.log(error);
        }
    }

    async loadData(){
        let data;
        try {
            data = await AsyncStorage.getItem('calendar');
            if (data !== null){
                data = JSON.parse(data);
                data.dateToday = new Date().toISOString().slice(0, 10);
                this.setState(data);
            } else{
                this.setState({
                    dateToday: new Date().toISOString().slice(0, 10),
                    children: [],
                    dateValue: '',
                    timeValue: '',
                    titleValue: '',
                    textValue: '',
                });
            }
        } catch (error) {
            console.log(error);
        }
    }

    createAppointment(e) {
        //Fetching values from form.
        let dateValue = e[0];
        let timeValue = e[1];
        let titleValue = e[2];
        let textValue = e[3];

        //Validate fields date, time and title. Last field is optional.
        if (this.validateFormDate(dateValue) && this.validateFormTime(timeValue) && this.validateFormTitle(titleValue)){

            //Creating new array with current state children.
            let newStateArray = this.state.children.slice();

            //Pushing new child to array.
            newStateArray.push({date: dateValue, time: timeValue, title: titleValue, text: textValue});

            //Sorting array with earliest appointments first.
            newStateArray.sort((a, b) => a.time > b.time);

            //Creating new updated state.
            let data = {
                children: newStateArray,
                dateToday: new Date().toISOString().slice(0, 10),
            };

            //Setting state.
            this.setState(data);

            //Setting updated values to storage.
            try{
                AsyncStorage.setItem("calendar", JSON.stringify(data));
            } catch (error){
                console.log(error);
            }

        } else {
            //If there are invalid values.
            alert('Error! Do not submit invalid values.')
        }
    }

    validateFormDate(date){
        return (date.length === 10)
    }

    validateFormTime(time){
        return (time.length === 5)
    }

    validateFormTitle(title){
        return (title.length > 0)
    }

    render(){
        return(
            <View style={styles.container}>
                <Week change={this.changeContent} />
                <View style={styles.bottomContainer}>
                    <AppointmentForm getValues={ arr => this.createAppointment(arr) } />
                    {this.emptyScheduleCheck()}
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        marginTop: 40,
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        marginBottom: 20,
    },
    bottomContainer: {
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
    },
    showAppointments: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
    },
    appointmentItem: {
        flexBasis:'33%',
        textAlign:'center',
        fontSize:16,
        flexWrap: 'wrap',
    },
    appointmentNoItems: {
        marginTop: '10%',
        flexBasis:'90%',
        textAlign:'center',
        fontSize:20,
    },
    scrollView: {
        height: '100%',
    }


});



