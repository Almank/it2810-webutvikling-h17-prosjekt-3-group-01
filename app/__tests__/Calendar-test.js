/**
 * Sample React Native Snapshot Test
 */
'use strict';

import 'react-native';
import React from 'react';
import {Calendar} from '../src/containers/Calendar';
import {AppointmentForm} from '../src/components/Calendar/AppointmentForm';
import {ContentHeader} from '../src/components/Calendar/ContentHeader';
import {Day} from '../src/components/Calendar/Day';
import {Week} from '../src/components/Calendar/Week';


const wrapper = shallow(<Calendar />);
const instWrapper = wrapper.instance();
//test that the calendar view and corresponding components renders correctly as well as testing functionality regarding emptyschedule check, creating appointments and todays date
describe('<Calendar />', () => {
	//test if the view renders correctly using snapshot
	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
	//check if appointmentForm renders correctly
	describe('<AppointmentForm />', () => {
		it('should render correctly', () => {
			const wrapper = shallow(<AppointmentForm />);
			expect(wrapper).toMatchSnapshot();
		});
	});
	//check if contentHeader renders correctly
	describe('<ContentHeader />', () => {
		it('should render correctly', () => {
			const wrapper = shallow(<ContentHeader />);
			expect(wrapper).toMatchSnapshot();
		});
	});
	//check if Week renders correctly
	describe('<Week />', () => {
		it('should render correctly', () => {
			const wrapper = shallow(<Week />);
			expect(wrapper).toMatchSnapshot();
		});
	});
	//test if the calendar's emptyScheduleCheck works
	describe('Calendar.emptyScheduleCheck()',() => {
		it('Should not be more then 0 events', () => {
			instWrapper.emptyScheduleCheck();
			const amountOfEvents = instWrapper.state.children.length;
			expect(amountOfEvents).toBe(0);
		});
	});
	//test if the calendar's createAppointment works
	describe('Calendar.createAppointment()',() => {
		it('should add one component', () => {
			instWrapper.createAppointment(['2017-10-15', '14:15', 'Title', 'Description']);
			const amountOfEvents = instWrapper.state.children.length;
			expect(amountOfEvents).toBe(1);
		});
	});
	//test if the calendars current date is todays date
	describe('Calendar.state.dateToday', () => {
		it('should match todays date', () => {
			expect(instWrapper.state.dateToday).toBe(new Date().toISOString().slice(0, 10));
		});
	});

});






