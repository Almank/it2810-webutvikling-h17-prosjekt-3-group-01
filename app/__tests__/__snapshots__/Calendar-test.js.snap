// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> <AppointmentForm /> should render correctly 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FF9505",
          "borderColor": "rgba(155,155,155,0.5)",
          "borderWidth": 1,
          "display": "flex",
          "flexBasis": 40,
          "justifyContent": "center",
          "marginLeft": "15%",
          "marginRight": "15%",
          "marginTop": 20,
        },
        Object {
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 5,
        },
      ]
    }
    title="Press me"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
      ADD APPOINTMENT
    </Text>
  </TouchableOpacity>
  <ContentHeader />
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Create new appointment
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        htmlFor="date"
        style={
          Object {
            "marginBottom": 3,
            "marginLeft": "15%",
            "marginRight": "15%",
            "marginTop": 10,
          }
        }
      >
        Date - YYYY-MM-DD
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="YYYY-MM-DD"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "rgba(155,155,155,0.5)",
              "borderWidth": 1,
              "flexBasis": 40,
              "marginLeft": "15%",
              "marginRight": "15%",
              "padding": 5,
            },
            Object {
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
            },
          ]
        }
        value=""
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        htmlFor="time"
        style={
          Object {
            "marginBottom": 3,
            "marginLeft": "15%",
            "marginRight": "15%",
            "marginTop": 10,
          }
        }
      >
        Time - HH:MM
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="HH:MM"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "rgba(155,155,155,0.5)",
              "borderWidth": 1,
              "flexBasis": 40,
              "marginLeft": "15%",
              "marginRight": "15%",
              "padding": 5,
            },
            Object {
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
            },
          ]
        }
        value=""
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        htmlFor="title"
        style={
          Object {
            "marginBottom": 3,
            "marginLeft": "15%",
            "marginRight": "15%",
            "marginTop": 10,
          }
        }
      >
        Title
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "rgba(155,155,155,0.5)",
              "borderWidth": 1,
              "flexBasis": 40,
              "marginLeft": "15%",
              "marginRight": "15%",
              "padding": 5,
            },
            Object {
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
            },
          ]
        }
        value=""
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        htmlFor="what"
        style={
          Object {
            "marginBottom": 3,
            "marginLeft": "15%",
            "marginRight": "15%",
            "marginTop": 10,
          }
        }
      >
        What
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "rgba(155,155,155,0.5)",
              "borderWidth": 1,
              "flexBasis": 40,
              "marginLeft": "15%",
              "marginRight": "15%",
              "padding": 5,
            },
            Object {
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
            },
          ]
        }
        value=""
      />
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF9505",
              "borderColor": "rgba(155,155,155,0.5)",
              "borderWidth": 1,
              "display": "flex",
              "flexBasis": 40,
              "justifyContent": "center",
              "marginLeft": "15%",
              "marginRight": "15%",
              "marginTop": 20,
            },
            Object {
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
            },
          ]
        }
        title="Submit"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 16,
            }
          }
        >
          Submit
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF9505",
              "borderColor": "rgba(155,155,155,0.5)",
              "borderWidth": 1,
              "display": "flex",
              "flexBasis": 40,
              "justifyContent": "center",
              "marginLeft": "15%",
              "marginRight": "15%",
              "marginTop": 20,
            },
            Object {
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 16,
            }
          }
        >
          Cancel
        </Text>
      </TouchableOpacity>
    </View>
  </Component>
</View>
`;

exports[`<Calendar /> <ContentHeader /> should render correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flexBasis": "33%",
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    When
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flexBasis": "33%",
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flexBasis": "33%",
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    About
  </Text>
</View>
`;

exports[`<Calendar /> <Week /> should render correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <Day
    dateFull="2017-10-17"
    dayDate={17}
    dayName="TODAY"
    key="2017-10-17"
  />
  <Day
    dateFull="2017-10-18"
    dayDate={18}
    dayName="TOMORROW"
    key="2017-10-18"
  />
  <Day
    dateFull="2017-10-19"
    dayDate={19}
    dayName="THURSDAY"
    key="2017-10-19"
  />
  <Day
    dateFull="2017-10-20"
    dayDate={20}
    dayName="FRIDAY"
    key="2017-10-20"
  />
  <Day
    dateFull="2017-10-21"
    dayDate={21}
    dayName="SATURDAY"
    key="2017-10-21"
  />
  <Day
    dateFull="2017-10-22"
    dayDate={22}
    dayName="SUNDAY"
    key="2017-10-22"
  />
  <Day
    dateFull="2017-10-23"
    dayDate={23}
    dayName="MONDAY"
    key="2017-10-23"
  />
</View>
`;

exports[`<Calendar /> should render correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "marginBottom": 20,
      "marginTop": 40,
    }
  }
>
  <Week
    change={[Function]}
  />
  <View
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <AppointmentForm
      getValues={[Function]}
    />
    <ScrollViewMock
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <View
        key="0"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "flexBasis": "90%",
              "fontSize": 20,
              "marginTop": "10%",
              "textAlign": "center",
            }
          }
        >
          You don't have any plans!
        </Text>
      </View>
    </ScrollViewMock>
  </View>
</View>
`;
